Tests/NIOTests/CircularBufferTests.swift

```diff
@@ -125,12 +136,29 @@ class CircularBufferTests: XCTestCase {
         XCTAssertEqual(ring.endIndex, 5)
 
         XCTAssertEqual(ring.index(after: 1), 2)
-
+        XCTAssertEqual(ring.index(before: 3), 2)
+        
         let actualValues = [Int](ring)
         let expectedValues = [0, 1, 2, 3, 4]
         XCTAssertEqual(expectedValues, actualValues)
     }
 
+    func testReplacingSubrangeOfElements() {
+        var ring = CircularBuffer<Int>(initialRingCapacity: 4)
+        for idx in 0..<50 {
+            ring.prepend(idx)
+        }
+        XCTAssertEqual(50, ring.count)
+        ring.replaceSubrange(20..<25, with: [99])
+
+        XCTAssertEqual(ring.count, 46)
+        XCTAssertEqual(ring[19], 30)
+        XCTAssertEqual(ring[20], 99)
+        XCTAssertEqual(ring[21], 24)
+        ring.replaceSubrange(ring.startIndex..<ring.endIndex, with: [])
+        XCTAssertTrue(ring.isEmpty)
+    }
```

Comment:
Can this test be extended to also test `replaceSubrange` with a subrange that is *larger* than the target range, and one that is equal to the subrange? Would be good to cover all the cases.